### PR TITLE
[Device]Support npu

### PR DESCRIPTION
**Pull request by flybird11111**
## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [ ] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [ ] I have added relevant tags if possible for us to better distinguish different PRs
- [ ] I have installed pre-commit: `pip install pre-commit && pre-commit install`

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [ ] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [ ] I have performed a self-review of my code
- [ ] I have added thorough tests.
- [ ] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [ ] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: applications/Colossal-LLaMA/colossal_llama/dataset/conversation.py [hunk 1]
```diff
@@ -100,7 +100,7 @@ def dict(self):
     messages=[],
     offset=0,
     sep_style=SeparatorStyle.ADD_BOS_EOS_TOKEN,
-    seps=["<|begin_of_text|>", "<|end_of_text|>"],
+    seps=["<|begin_of_text|>", "<|eot_id|>"],
```
Comment (ver217) [inline applications/Colossal-LLaMA/colossal_llama/dataset/conversation.py:103]: @TongLi3701 Please take a look. Is this right?

File: colossalai/pipeline/schedule/interleaved_pp.py [hunk 2]
```diff
@@ -18,7 +17,7 @@
 from .base import PipelineSchedule
 
 
-def _wait_p2p(wait_handles: List[torch.cuda.Event]) -> None:
+def _wait_p2p(wait_handles: List[get_accelerator().Event]) -> None:
```
Comment (ver217) [inline colossalai/pipeline/schedule/interleaved_pp.py:20]: `get_accelerator().Event` is incomplete type during coding time

File: colossalai/shardformer/layer/normalization.py [hunk 3]
```diff
@@ -1,15 +1,29 @@
...
 from .utils import SeqParallelUtils
 
+SUPPORT_NPU = False
+try:
+    import torch_npu
+
+    SUPPORT_NPU = True
+    warnings.warn("support npu")
+except Exception:
+    warnings.warn("support gpu")
```
Comment (ver217) [inline colossalai/shardformer/layer/normalization.py:24]: No need to warn

File: examples/language/llama/benchmark.py [hunk 4]
```diff
@@ -337,7 +340,7 @@ def empty_init():
 
     torch.set_default_dtype(torch.float)
     coordinator.print_on_master(
-        f"Booster init max CUDA memory: {get_accelerator().max_memory_allocated()/1024**2:.2f} MB"
+        f"Booster init max NPU memory: {get_accelerator().max_memory_allocated()/1024**2:.2f} MB"
```
Comment (ver217) [inline examples/language/llama/benchmark.py:343]: device memory

File: extensions/pybind/flash_attention/flash_attention_npu.py [hunk 5]
```diff
@@ -27,6 +27,7 @@ def build_jit(self) -> None:
         )
 
     def load(self):
+        import math
```
Comment (ver217) [inline extensions/pybind/flash_attention/flash_attention_npu.py:30]: move it to top level

File: applications/Colossal-LLaMA/colossal_llama/dataset/conversation.py (hunk 1 above)
Comment (TongLi3701) [reply applications/Colossal-LLaMA/colossal_llama/dataset/conversation.py:103]: It should be ok for now.

File: colossalai/pipeline/schedule/interleaved_pp.py (hunk 2 above)
Comment (flybird11111) [reply colossalai/pipeline/schedule/interleaved_pp.py]: updated.

File: colossalai/shardformer/layer/normalization.py (hunk 3 above)
Comment (flybird11111) [reply colossalai/shardformer/layer/normalization.py]: updated.

File: extensions/pybind/flash_attention/flash_attention_npu.py (hunk 5 above)
Comment (flybird11111) [reply extensions/pybind/flash_attention/flash_attention_npu.py]: updated.

File: examples/language/llama/benchmark.py (hunk 4 above)
Comment (flybird11111) [reply examples/language/llama/benchmark.py]: updated

File: examples/language/llama/benchmark.py [hunk 6]
```diff
@@ -389,7 +392,7 @@ def empty_init():
                 performance_evaluator.on_step_end(**batch)
                 prof.step()
     performance_evaluator.on_fit_end()
-    coordinator.print_on_master(f"Max CUDA memory usage: {get_accelerator().max_memory_allocated()/1024**2:.2f} MB")
+    coordinator.print_on_master(f"Max NPU memory usage: {get_accelerator().max_memory_allocated()/1024**2:.2f} MB")
```
Comment (ver217) [inline examples/language/llama/benchmark.py:395]: same problem

Comment (flybird11111) [reply examples/language/llama/benchmark.py]: updated.

**Review by ver217 — APPROVED** [review]
